doubly_linked_list/doubly_linked_list.py

```diff
@@ -44,25 +44,118 @@ def __len__(self):
     return self.length
 
   def add_to_head(self, value):
-    pass
+    if not self.head:
+      self.head = ListNode(value)
+      self.tail = self.head
+      self.length = 1
+      return
+    
+    # self.old_head = self.head
+    # self.new_head = ListNode(value)
+    # self.new_head.next = self.old_head
+    # self.old_head.prev = self.new_head
+    # self.head = self.new_head
+
+    self.head.insert_before(value)
+    self.head = self.head.prev
+    self.length += 1
 
```

Comment:
this looks interesting; would need you to walk me through it.
are you taking into account a previous head that needs to be reconfigured to not be a head anymore but still connect to the new head and the next node?
where is **insert_before()** coming from?

Reply:
my mistake on the insert_before()